### ISSUE TITLE
Where to place configuration?

### ISSUE THREAD

**Issue by BnMcG** [sensu-plugins/sensu-plugins-mailer#15]
I'm having trouble working out where to put the configuration. So far I've tried putting a mailer.json with a "mailer-mailgun" section in /etc/sensu/, /etc/sensu/conf.d and also placing the "mailer-mailgun" section directly into the handler definition like so:

`
{
  "handlers": {
    "mailer-mailgun": {
      "type": "pipe",
      "command": "/etc/sensu/plugins/sensu-plugins-mailer/bin/handler-mailer-mailgun.rb"
    }
  },
  "mailer-mailgun": {
    "mail_from": "monitoring@bla.com",
    "mail_to": "support@bla.com",
    "mg_apikey": "bla",
    "mg_domain": "mg.bla.com",
    "subscriptions": {
      "blabla": {
        "mail_to": "bla2@gmail.com"
      }
  }
}
`

I keep receiving the following error:

> {"timestamp":"2016-03-03T05:50:14.375408-0500","level":"info","message":"handler output","handler":{"type":"pipe","command":"/etc/sensu/plugins/sensu-plugins-mailer/bin/handler-mailer-mailgun.rb","name":"mailer-mailgun"},"output":["/etc/sensu/plugins/sensu-plugins-mailer/bin/handler-mailer-mailgun.rb:46:in `handle': undefined method`[]' for nil:NilClass (NoMethodError)\n","\tfrom /var/lib/gems/1.9.1/gems/sensu-plugin-1.2.0/lib/sensu-handler.rb:55:in `block in class:Handler'\n"]}

Which makes me think it can't find the configuration. Where does it need to be placed so that the plugin can find it?

**Comment by bgaillard** [comment]
Hi @BnMcG, I encountered exactly the same problem, finaly it seems that everything has to be placed inside your Handler configuration file.

For example in my case I created a `/etc/sensu/conf.d/mailer_handler.json` configuration file with the following content.

```json
{
    "mailer" :  {
        "admin_gui" : "http://192.168.99.100:3000",
        "mail_from": "sensu@7opteam.com",
        "mail_to": "baptiste.gaillard@gomoob.com",
        "delivery_method": "smtp",
        "smtp_address": "smtp.gmail.com",
        "smtp_port": "587",
        "smtp_domain": "gmail.com",
        "smtp_enable_starttls_auto": "true",
        "smtp_username" : "baptiste.gaillard@gomoob.com",
        "smtp_password" : "XXXXXXXX"
    },
    "handlers": {
        "mailer": {
            "type": "pipe",
            "command": "/usr/local/bin/handler-mailer.rb"
        }
    }
}
```

I searched several hours to find this here https://github.com/sensu/sensu-community-plugins/issues/852 and think the README.md has to be updated to add much more details to explain how to configure the plugin.

**Comment by BnMcG** [comment]
Cheers @bgaillard - I'll give this a go soon. Thanks very much for letting me know about this!

**Comment by bgaillard** [comment]
> Cheers @bgaillard - I'll give this a go soon. Thanks very much for letting me know about this!

@BnMcG :+1: thanks.

**Comment by BnMcG** [comment]
For reference, this is the configuration that worked for me for Mailgun alerts:

In the file: /etc/sensu/conf.d/handlers/mailer-mailgun.json:

`
{
    "mailer-mailgun" :  {
        "mail_from": "monitoring@mydomain.com",
        "mail_to": "support@mydomain.com",
        "mg_apikey": "key-mykey",
        "mg_domain": "mg.mydomain.com"
    },
    "handlers": {
        "mailer-mailgun": {
            "type": "pipe",
            "command": "/opt/sensu/embedded/bin/ruby /etc/sensu/plugins/sensu-plugins-mailer/bin/mailer-mailgun.rb"
        }
    }
}
`

**Comment by majormoses** [comment]
closed via #50
